### PR TITLE
Block certain actions when query is not valid

### LINKED ISSUE #1345: Feature request: Block certain actions when query is not valid
### Similar requests

- [X] I have searched and found no existing similar requests

### How are you using Studio?

Legend Studio

### What problems are you trying to solve?

Block certain actions when query is no valid

### Describe the solution you would like to see implemented

Avoid users to run options in result panel when
- [x] They have duplicated columns in projection panel
- [x] There are no columns in the projection panel
- [x] Do not disable run options when `graphFetch` is selected

We already handle the case where user tries to create duplicate parameters in form mode. We diable the `create` button in this scenario. We don't want to handle the case where user creates duplicate parameters in text mode.

### Describe alternatives you have considered

_No response_

### Documentation, Design, Adoption, Migration Strategy

_No response_

### Contribution

- [X] I would like to work on this feature

### PR DESCRIPTION
**Pull request by gayathrir11**
## Summary

closes #1345 

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

https://user-images.githubusercontent.com/87973126/182402667-672d26ed-3c85-4172-a39d-8d3732b4d09f.mp4

### CODE REVIEW

**Comment by YannanGao-gs** [comment]
@gayathrir11  What's the current behavior if users want to set the filter value to an empty string?

File: packages/legend-application-query/src/components/QueryBuilderFilterPanel.tsx [hunk 1]
```diff
@@ -214,6 +217,9 @@ const QueryBuilderFilterConditionEditor = observer(
         node.condition.operator.getDefaultFilterConditionValue(node.condition),
       );
     };
+    const isEmptyNode = (): boolean =>
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/components/QueryBuilderFilterPanel.tsx:220]: Empty string values are valid inputs.

File: packages/legend-application-query/src/stores/QueryBuilderProjectionState.ts [hunk 2]
```diff
@@ -793,4 +793,17 @@ export class QueryBuilderProjectionState {
       );
     }
   }
+
+  isValidProjectionState(): boolean {
+    const hasDuplicatedProjectionColumns = this.columns.some(
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/stores/QueryBuilderProjectionState.ts:798]: lets make this a bit more readable. 
We can do something like 
```
if(this.queryBuilderState.fetchStructureState.isProjectionMode)
{
   // duplicate cols check
  // no cols check 
}
return true;
````

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-application-query/src/stores/QueryBuilderProjectionState.ts [hunk 3]
```diff
@@ -793,4 +793,19 @@ export class QueryBuilderProjectionState {
       );
     }
   }
+
+  isValidProjectionState(): boolean {
+    if (this.queryBuilderState.fetchStructureState.isProjectionMode()) {
+      // duplicate columns check
+      const hasDuplicatedProjectionColumns = this.columns.some(
+        (column) =>
+          this.columns.filter((c) => c.columnName === column.columnName)
```
Comment (YannanGao-gs) [inline packages/legend-application-query/src/stores/QueryBuilderProjectionState.ts:802]: the time complexity of duplication check is O(n^2) 
too high

File: packages/legend-application-query/src/stores/QueryBuilderFilterState.ts [hunk 4]
```diff
@@ -415,6 +415,7 @@ export class QueryBuilderFilterState
     makeAutoObservable(this, {
       queryBuilderState: false,
       isValidMove: false,
+      nodes: observable,
```
Comment (MauricioUyaguari) [inline packages/legend-application-query/src/stores/QueryBuilderFilterState.ts:418]: why do we need this?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
why do we need this?

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**